lib/worker.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+var Promise = require('bluebird');
+var TaskFatalError = require('./errors/task-fatal-error');
+var TaskMaxRetriesError = require('./errors/task-max-retries-error');
+var assign = require('101/assign');
+var defaults = require('101/defaults');
+var error = require('./error');
+var exists = require('101/exists');
+var log = require('./logger').child({ module: 'worker' });
+var pick = require('101/pick');
+
+/**
+ * Worker module, for performing job tasks.
+ * @author Bryan Kendall
+ * @author Ryan Sandor Richards
+ * @module ponos
+ */
+module.exports = Worker;
+
+/**
+ * Worker class: performs tasks for jobs on a given queue.
+ * @class
+ * @param {object} opts Options for the worker
+ * @param {string} opts.queue Name of the queue for the job the worker
+ *   is processing.
+ * @param {function} opts.task A function to handle the tasks
+ * @param {object} opts.job Data for the job to process.
+ * @param {function} opts.done Callback to execute when the job has successfully
+ *   been completed.
+ * @param {boolean} [opts.runNow] Whether or not to run the job immediately,
+ *   defaults to `true`.
+ */
+function Worker (opts) {
+  // managed required fields
+  var fields = [
+    'queue',
+    'task',
+    'job',
+    'done'
+  ];
+  fields.forEach(function (f) {
+    if (!exists(opts[f])) {
+      throw new Error(f + ' is required for a Worker');
+    }
+  });
+
+  // manage field defaults
+  fields.push('runNow');
```

Comment:
mostly for testing, so you can create a worker in a before and then run it in the test